var/spack/repos/builtin/packages/intel-mpi/package.py

```diff
@@ -83,7 +83,14 @@ def setup_dependent_package(self, module, dep_spec):
         else:
             raise RuntimeError('No suitable bindir found')
 
-        self.spec.mpicc = join_path(bindir, 'mpicc')
```

Comment:
It worked for me ok on some test build a few months ago.

Here's the relevant discussion why the code is correct and tested package: https://github.com/LLNL/spack/pull/3905